### PR TITLE
Add missing team stream route for user by track

### PR DESCRIPTION
**Pull request by yannvery**
### Description

If you try to open a team stream on a specifi user for instance : 
[/teams/baconesia/streams/users/beth](http://localhost:4567/teams/baconesia/streams/users/beth) you'll see that the last part of the sidebar contains a list of links. One link for each track the user's worked on.

For example:
[/teams/baconesia/streams/users/beth/tracks/coffeescript](http://localhost:4567/teams/baconesia/streams/users/beth/tracks/coffeescript)

This link is built by a `TeamStream::UserFilter` object but the relative url this object returns does not match with an existing route.

### Solution
Add the relative route on `app/routes/team.rb`file

### CODE REVIEW

File: app/routes/teams.rb [hunk 1]
```diff
@@ -156,6 +156,35 @@ class Teams < Core
...
 
+      get '/teams/:slug/streams/users/:username/tracks/:id' do |slug, username, track_id|
+        please_login
+        only_with_existing_team(slug) do |team|
+          unless team.includes?(current_user)
+            flash[:error] = "You may only view team pages for teams that you are a member of, or that you manage."
+            redirect '/'
+          end
+
+          user = ::User.find_by_username(username)
```
Comment (bernardoamc) [inline app/routes/teams.rb]: Nowadays it's recommended to use `User.find_by(username: username)` instead of dynamic finders. :)

**Comment by bernardoamc** [comment]
Thank you for the PR @yannvery! <3

Should we adjust the tests to take into account this route? We have a few existing tests [here](https://github.com/exercism/exercism.io/blob/master/test/app/teams_test.rb) for teams.

File: app/routes/teams.rb [hunk 2]
```diff
@@ -156,6 +156,35 @@ class Teams < Core
...
+          stream = TeamStream.new(team, current_user.id, params['page'] || 1)
+          stream.track_id = track_id.downcase
```
Comment (bernardoamc) [inline app/routes/teams.rb]: No need to `downcase` here, the `writer` already [takes care of it](https://github.com/exercism/exercism.io/blob/master/lib/exercism/team_stream.rb#L19).

**Comment by yannvery** [comment]
Thanks for the review @bernardoamc !
I've added tests for all team stream routes, let me know if they need to be written another way.

File: test/app/teams_test.rb [hunk 3]
```diff
@@ -356,4 +356,73 @@ def test_cannot_disown_team_when_sole_manager
...
+    team.save
+    TeamMembershipInvite.pending.find_by(user: charlie, team: team).accept!
+
+    # Charlie has submitted one exercise, but Bob is not the same team.
+    create_exercise_with_submission(charlie, 'fake', 'hello-world')
+
+    get '/teams/awesome/streams', {}, login(bob)
+    assert_response_status 302
+    assert_equal "http://example.org/", last_response.location
+
```
Comment (bernardoamc) [inline test/app/teams_test.rb]: ✂️

File: test/app/teams_test.rb [hunk 4]
```diff
@@ -356,4 +356,73 @@ def test_cannot_disown_team_when_sole_manager
...
+    assert_response_status 200
+
+    get '/teams/awesome/streams/users/charlie', {}, login(bob)
+    assert_response_status 200
+
+    get '/teams/awesome/streams/users/charlie/tracks/fake', {}, login(bob)
+    assert_response_status 200
+  end
+
+  def create_exercise_with_submission(user, language, slug, ts=1.day.ago)
```
Comment (bernardoamc) [inline test/app/teams_test.rb]: Can you add a `private` before this method? :)

File: test/app/teams_test.rb [hunk 5]
```diff
@@ -356,4 +356,73 @@ def test_cannot_disown_team_when_sole_manager
...
+      user_exercise_id: exercise.id,
+      language: exercise.language,
+      slug: exercise.slug,
+      created_at: ts,
+      updated_at: ts,
+    }
+    Submission.create!(attributes)
+  end
 end
+
```
Comment (bernardoamc) [inline test/app/teams_test.rb]: ✂️

File: test/app/teams_test.rb (hunk 3 above)
Comment (yannvery) [inline test/app/teams_test.rb]: I've assumed you pointed the blank line :)

Comment (bernardoamc) [inline test/app/teams_test.rb]: Haha yes 😄

**Comment by kytrinyx** [comment]
This is great. Thanks so much for taking the time to fix it (including tests! Much appreciated). Thanks @bernardoamc for the detailed reviews along the way.

**Pull request merged**